### PR TITLE
 Zarr: add MULTIBAND=YES/NO, DIM_X and DIM_Y open options (fixes #8237) 

### LINKED ISSUE #8237: gdal_translate zarr to multi-band tif
After finding an [example in the docs]((https://gdal.org/drivers/raster/zarr.html#:~:text=array%20to%20GeoTIFF%3A-,gdal_translate,-%27ZARR%3A%22my.zarr)), I found that multi-band zarr to tif is not working i.e. `gdal_translate data.zarr data.tif`.
For now the python alternative works fine, but a cli option for this is defo a nice to have.

creating a 3d zarr file from a few tifs
```
import rioxarray as rxr
import xarray as xr

s1 = rxr.open_rasterio("s1.tif", chunks=True)
s2 = rxr.open_rasterio("s2.tif", chunks=True)
ds = xr.concat([s1, s2], dim="band")
ds.to_zarr("data.zarr")
```

converting the zarr to a multi-band tif
```
ds_zarr = xr.open_zarr("data.zarr").__xarray_dataarray_variable__
ds_zarr.rio.to_raster("data.tif")
```

To handle the flexibility of zarr data, the gdal_translate could require the coordinates [x, y, band].

### PR DESCRIPTION
**Pull request by rouault**
and default to MULTIBAND=YES

### CODE REVIEW

File: frmts/zarr/zarrdriver.cpp [hunk 1]
```diff
@@ -1443,6 +1573,14 @@ void GDALRegister_Zarr()
         "   <Option name='CACHE_TILE_PRESENCE' type='boolean' "
         "description='Whether to establish an initial listing of present "
         "tiles' default='NO'/>"
+        "   <Option name='MULTIBAND' type='boolean' default='NO' "
+        "description='Whether to expose >= 3D arrays as GDAL multiband "
```
Comment (jratike80) [inline frmts/zarr/zarrdriver.cpp:1577]: Should it be "default='YES'" like in the zarr.rst above?

Comment (rouault) [reply frmts/zarr/zarrdriver.cpp:1577]: good point . fixed

**Comment by mdsumner** [comment]
thanks Even!  

fwiw initial test with my go-to Zarr-url is very slow

```
gdalinfo --config AWS_NO_SIGN_REQUEST YES ZARR:"/vsis3/mur-sst/zarr-v1/":/analysed_sst -oo MULTIBAND=YES

gdal_translate --config AWS_NO_SIGN_REQUEST YES ZARR:"/vsis3/mur-sst/zarr-v1/":/analysed_sst -oo MULTIBAND=YES -b 1 out.tif
```

I appreciate it could just be slow because my compute is not in the right S3 tenancy. 

(setting MULTIBAND=NO precludes that syntax as expected)

with a local source  it's fast and behaves as expected (MULTIBAND and the DIM_X/DIM_Y), I'm hoping to put more time into this to explore the various options

**Comment by rouault** [comment]
> I appreciate it could just be slow because my compute is not in the right S3 tenancy.

the reason is that the ``time`` variable has shape = [6443] but chunks=[5]. Consequently, when instantiating the GDAL dataset and assign to each band a DIM_time_VALUE metadata item, we need to read 6443 / 5 chunk data files of the time variable...

**Comment by mdsumner** [comment]
ah I see, thank you

**Comment by rouault** [comment]
> the reason is that the `time` variable has shape = [6443] but chunks=[5]. Consequently, when instantiating the GDAL dataset and assign to each band a DIM_time_VALUE metadata item, we need to read 6443 / 5 chunk data files of the time variable...

I've just added a LOAD_EXTRA_DIM_METADATA_DELAY option to offer a configurable maximum delay, that defaults to 5 sec

**Comment by mdsumner** [comment]
excellent set of fixes, this is awesome

**Pull request merged**